test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -1353,6 +1411,436 @@ TEST_F(OpenTelemetryPluginNPCMetricsTest,
+TEST_F(OpenTelemetryPluginNPCMetricsTest, ThreadedSetInt64Gauge) {
+  constexpr absl::string_view kMetricName = "yet_another_int64_gauge";
+  constexpr std::array<absl::string_view, 2> kLabelKeys = {"label_key_1",
+                                                           "label_key_2"};
+  constexpr std::array<absl::string_view, 2> kOptionalLabelKeys = {
+      "optional_label_key_1", "optional_label_key_2"};
+  constexpr std::array<absl::string_view, 2> kLabelValues = {"label_value_1",
+                                                             "label_value_2"};
+  constexpr std::array<absl::string_view, 2> kOptionalLabelValues = {
+      "optional_label_value_1", "optional_label_value_2"};
+  auto handle = grpc_core::GlobalInstrumentsRegistry::RegisterInt64Gauge(
+      kMetricName, "A simple int64 gauge.", "unit", kLabelKeys,
+      kOptionalLabelKeys,
+      /*enable_by_default=*/true);
+  Init(std::move(Options()
+                     .set_metric_names({kMetricName})
+                     .add_optional_label(kOptionalLabelKeys[0])
+                     .add_optional_label(kOptionalLabelKeys[1])));
+  ThreadedMetricsCollector collector{
+      this, grpc_core::Duration::Zero(), -1,
+      [&](const absl::flat_hash_map<
+          std::string,
+          std::vector<opentelemetry::sdk::metrics::PointDataAttributes>>&
+              data) { return !data.contains(kMetricName); }};
+  auto stats_plugins =
+      grpc_core::GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
+          grpc_core::StatsPlugin::ChannelScope("dns:///localhost:8080", ""));
+  constexpr int kIterations = 1000000;
+  for (int i = 0; i < kIterations; ++i) {
+    stats_plugins.SetGauge(handle, i, kLabelValues, kOptionalLabelValues);
+  }
+  auto data = collector.Stop();
+  for (size_t i = 1; i < data[kMetricName].size(); ++i) {
+    EXPECT_THAT(data[kMetricName][i],
+                ::testing::AllOf(
+                    GaugeResultGe(opentelemetry::nostd::get<int64_t>(
+                        opentelemetry::nostd::get<
+                            opentelemetry::sdk::metrics::LastValuePointData>(
+                            data[kMetricName][i - 1].point_data)
+                            .value_)),
+                    GaugeResultLaterThan(
+                        opentelemetry::nostd::get<
+                            opentelemetry::sdk::metrics::LastValuePointData>(
+                            data[kMetricName][i - 1].point_data)
+                            .sample_ts_)));
+  }
+  // Verify labels.
+  EXPECT_THAT(data,
+              ::testing::UnorderedElementsAre(::testing::Pair(
+                  kMetricName, ::testing::Each(AttributesEq(
+                                   kLabelKeys, kLabelValues, kOptionalLabelKeys,
+                                   kOptionalLabelValues)))));
+}
+
+using OpenTelemetryPluginCallbackMetricsTest = OpenTelemetryPluginEnd2EndTest;
+
+// The report duration is longer than the collect duration, so we expect to
```

Comment:
I know what you intended here, but the terminology isn't clear: it's not obvious what "report duration" and "collect duration" refer to.  Suggest instead saying "OT reporting interval" and "callback min interval".

Reply:
Done.